### ISSUE TITLE
Dust3D

### ISSUE THREAD

**Issue by tex** [NixOS/nixpkgs#90231]
**Project description**

Making 3D Model Has Never Been So Easy
Auto UV unwrapping, auto rigging with PBR Material support, pose and motion authoring all in one. 3D Modeling in a blink, try it now!

**Metadata**

* homepage URL: https://dust3d.org/
* source URL: https://github.com/huxingyi/dust3d
* license: Dust3D's source code is available under the MIT license, however, some third party libraries we use are not MIT license compatible, such as CGAL. You can use the released Dust3D software freely no matter for personal or for commercial purpose, and you can integrate the source code of Dust3D to your project as MIT licensed, but with the third party libraries on the specified license respectively.
* platforms: linux, darwin, windows

**Comment by tex** [comment]
Very usefull to easily create models that can be exported into Sweet Home 3D...

**Comment by Thra11** [comment]
CGAL shouldn't be an issue for people wanting to use the program, as the parts are either dual licensed GPL+commercial or LGPL+commercial, so the combination of dust3d and cgal can be used under GPL. It's also already packages in nixpkgs.

The weirdly licensed dependency is actually instant-meshes, which appears to be BSD 3-clause with an added paragraph.

**Comment by Thra11** [comment]
I had a quick look at packaging this. Frankly, it's a massive pain, as many of the dependencies are "vendored", for example, instant-meshes doesn't build a static or shared library which can be linked against by dust3d, so dust3d needs the entire instant-meshes source tree in a subdirectory. instant-meshes itself has at least six vendored dependencies of its own, with no option to use system libraries instead, even if they exist.

**Comment by tex** [comment]
Yeah, seems complicated. They also provide appimage, is it an option to use this in nixpkgs? I tested it and it works, well, I guess it is missing icons, but I am not sure as I use i3 and I am not used to see much icons anyway :)
